library-valcon-internship/src/pages/HomePage/HomePage.tsx

```diff
@@ -1,12 +1,49 @@
+import { useEffect, useState } from 'react'
+
+import InfiniteScroll from 'react-infinite-scroll-component'
 import { NavLink } from 'react-router-dom'
 
 import addIcon from '../../assets/icons/add-icon.svg'
+import BookList from '../../components/BookList/BookList'
+import Book from '../../models/Book'
+import { getBooks } from '../../services/BookService'
 import './HomePage.css'
 
 const HomePage = () => {
+  const [ pageNumber, setPageNumber ] = useState(1)
+  const [ books, setBooks ] = useState<Book[]>([])
+  const [ isEmptyResponse, setIsEmptyResponse ] = useState(false)
+  useEffect(() => {
+    getBooks(pageNumber, 10)
+      .then(response => {
+        setIsEmptyResponse(response.data.length === 0)
+        setBooks(prevBooks => [ ...prevBooks, ...response.data ])
+      })
+      .catch(() => {
+        setBooks([])
+      })
+  }, [ pageNumber ])
+  const handleNextPage = () => {
+    setPageNumber(prevPageNumber => prevPageNumber + 1)
+  }
   return (
     <div className='homePage'>
-      <h3>No books currently available</h3>
+      {
+        books ?
+          (
+            <InfiniteScroll
+              dataLength={books.length}
+              next={handleNextPage}
+              hasMore={!isEmptyResponse}
```

Comment:
How is has more false only if it's an empty response for previous request? That means we always have an extra request. Has more should be calculated based on total elements, current page and num of elements per page

Reply:
The request does not return the total number of elements so I cannot calculate if there are no more books without an additional request.